### ISSUE TITLE
[organization] stacks and folders are just collections

### ISSUE THREAD

**Issue by zhenyasav** [dxos/dxos#5166]
- underlying both stacks & folders are collections of objects
- stack is just one way to render a collection of objects (another would be grid)
- we cannot simply render a folder as a stack/grid though as they have extra view metadata
  - this metadata could have reasonable defaults, but if edited it needs to be persisted somewhere
- the expectation is to resolve this with a "data layer" above the ECHO db where different objects can be joined together into a single view
- with resolvers in a data layer a "stack" can be a joined view of a collection and stack view metadata (e.g. section heights)
- currently blocked by #5124 as that is what will form the data layer (which should probably be another issue as well)

**Comment by wittjosiah** [comment]
Blocked by #4745
